Makefile-az.mk

```diff
@@ -82,17 +91,17 @@ az-patch-skaffold: 	## Update Azure client env vars and settings in skaffold con
 	yq -i  '(.manifests.helm.releases[0].overrides.controller.env[] | select(.name=="KUBELET_BOOTSTRAP_TOKEN")).value =                             "$(BOOTSTRAP_TOKEN)"'         skaffold.yaml
 	yq -i  '(.manifests.helm.releases[0].overrides.controller.env[] | select(.name=="SSH_PUBLIC_KEY")).value =                                               "$(SSH_PUBLIC_KEY)"'          skaffold.yaml
 
-az-patch-skaffold-kubenet: az-patch-skaffold	az-fetch-network-info
-	$(eval AZURE_SUBNET_ID=$(shell az network vnet list --resource-group $(AZURE_RESOURCE_GROUP_MC) | jq  -r ".[0].subnets[0].id"))
+az-patch-skaffold-kubenet: az-patch-skaffold az-fetch-network-info
+	$(eval AZURE_SUBNET_ID=$(shell az network vnet list --resource-group $(AZURE_VNET_RESOURCE_GROUP) | jq  -r ".[0].subnets[0].id"))
 	yq -i '(.manifests.helm.releases[0].overrides.controller.env[] | select(.name=="AZURE_SUBNET_ID"))               .value = "$(AZURE_SUBNET_ID)"'         skaffold.yaml
 	yq -i  '(.manifests.helm.releases[0].overrides.controller.env[] | select(.name=="NETWORK_PLUGIN").value) =                                              "kubenet"'                    skaffold.yaml
 
-az-patch-skaffold-azure: az-patch-skaffold	az-fetch-network-info
+az-patch-skaffold-azure: az-patch-skaffold az-fetch-network-info
 	$(eval AZURE_SUBNET_ID=$(shell az aks show --name $(AZURE_CLUSTER_NAME) --resource-group $(AZURE_RESOURCE_GROUP) | jq -r ".agentPoolProfiles[0].vnetSubnetId"))
 	yq -i '(.manifests.helm.releases[0].overrides.controller.env[] | select(.name=="AZURE_SUBNET_ID"))               .value = "$(AZURE_SUBNET_ID)"'         skaffold.yaml
 
 az-patch-skaffold-azureoverlay: az-patch-skaffold	az-fetch-network-info
-	$(eval AZURE_SUBNET_ID=$(shell az network vnet list --resource-group $(AZURE_RESOURCE_GROUP_MC) | jq  -r ".[0].subnets[0].id"))
+	$(eval AZURE_SUBNET_ID=$(shell az network vnet list --resource-group $(AZURE_VNET_RESOURCE_GROUP) | jq  -r ".[0].subnets[0].id"))
```

Comment:
Can we move away from the shortcut of picking the first subnet of the first VNET observed, and actually get this from the cluster (or system nodepool)? Or is this not possible for self-hosted?

Reply:
this is possible just didnt want to change the pattern. It makes sense for self hosted but it does add a dependency on the agentpool which i thought we were trying to avoid.

Reply:
picked from agentpool